test/unit/errorResponseInterceptor.test.js

```diff
@@ -0,0 +1,419 @@
+// and convert it to a JSON object
+async function getJsonErrorFromStreamError(streamError) {
+  let data = '';
+  await pipeline(
+    streamError.result,
+    new Writable({
+      write: (c, e, cb) => {
+        data += c;
+        cb();
+      },
+    })
+  );
+  return JSON.parse(data);
+}
+
+describe('test errorResponseInterceptor', () => {
+  beforeEach(() => {
+    nock.cleanAll();
+    service = getClient(); // new client at each run
+  });
+
+  it.each(getCases())('$name', async (test) => {
+    // prepare request function name, response method, response code
+    const requestFunction = test.requestFunction
+      ? test.requestFunction
+      : 'getDocument';
+    const responseMethod = requestFunction.split(/(?=[A-Z])/)[0];
+    const responseCode = test.responseCode ? test.responseCode : ERROR_CODE;
+    // mock response
+    nock(DB_URL)
+      [responseMethod](`/${DOC_ID}`)
+      .reply(responseCode, test.mockResponse, test.mockHeaders);
+    let responseIsSaved = false;
+
+    // declare variables for assertions
+    let responseStatus;
+    let responseHeaders;
+    let responseResult;
+    let responseMessage;
+    let responseStack;
+
+    await service[requestFunction](DOCUMENT_REQUEST)
+      .then((response) => {
+        responseStatus = response.status;
+        responseHeaders = response.headers.toJSON();
+        responseResult = response.result;
+        responseIsSaved = true;
+      })
+      .catch(async (onrejected) => {
+        if (!responseIsSaved) {
+          responseStatus = onrejected.status;
+          if (onrejected.headers) {
+            responseHeaders = onrejected.headers.toJSON();
+          }
+          // Get JSON when result is a stream:
+          if (onrejected.result instanceof IncomingMessage) {
+            responseResult = await getJsonErrorFromStreamError(onrejected);
+          } else {
+            responseResult = onrejected.result;
+          }
+          responseMessage = onrejected.message;
+          responseStack = onrejected.stack;
+          responseIsSaved = true;
+        }
+      });
+    // response was saved in either the then or the catch block:
+    assert.ok(responseIsSaved);
+    if (test.expectedResponseCode === -1) {
+      // responseStatus is undefined for edge case test_no_augment_invalid_json:
+      assert.equal(responseStatus, undefined);
+    } else {
+      assert.equal(responseStatus, responseCode);
+    }
+    assert.deepEqual(responseHeaders, test.expectedHeaders);
+    assert.deepStrictEqual(responseResult, test.expectedResponse);
+    if (test.expectedResponseMessage) {
+      // check message includes SyntaxError for edge case test_no_augment_invalid_json:
+      assert.ok(responseMessage.includes(test.expectedResponseMessage));
+      assert.ok(!responseStack.includes('errorResponseInterceptor'));
+    }
+  });
+
+  it('test_added', async () => {
+    const expectedNumberOfResponseInterceptors = numberOfBaseInterceptors + 1;
```

Comment:
Since `numberOfBaseInterceptors = 1` the expected here is `2`.
AFAIK if retries are not also configured the error interceptor should be the only one, so I don't really understand why we're asserting `2`. (Except that possibly we're accidentally adding 2 when we shouldn't be because of the `newInstance`/`configureService` thing.

Reply:
I added one comment for this in 95ebee7b.

Reply:
I think I probably even did the PR for that at some point and forgetten about it.